torch/ao/quantization/_quantize_dynamic_tracing.py

```diff
@@ -0,0 +1,75 @@
+import torch
+
+from .dynamic_tracing.auto_trace import add_auto_observation, add_auto_convert
+from .dynamic_tracing.fusion import get_module_fusion_fqns
+
+
+def prepare(model, example_inputs, inplace=False, allow_list=None,
+            observer_non_leaf_module_list=None,
+            prepare_custom_config_dict=None,
+            fuse_modules=True):
+    r"""A wrapper around `torch.quantization.prepare` which prepares the
+    model for quantization using dynamic tracing. Requires `example_inputs` to build
+    the graph before calibration or quantization aware training can proceed.
+
+    TODO(future PR): better docblock
+    """
+    assert example_inputs is not None, 'example_inputs must be specified'
+
+    if fuse_modules:
```

Comment:
Given that our existing eager mode API does fusion separately, should fusing modules be the default?

Also, should we consolidate how we handle fusions? 

it looks like currently each quantization method does something different:

eager mode: fusion needs to be done in a separate function call before prepare
fx: fusion is mandatory in prepare
dynamic tracing: fusion is default in prepare with an option to turn it off

Reply:
In eager mode, fusion is a separate function because there is not a good way to automate it and it requires user input on modules to fuse.  In an automated approach such as this one, we can fuse modules and functions automatically.  This is why I'd recommend a "fusion on by default" API.

Sure, we can look into making fusion mandatory in this API, to match with the FX graph mode quantization API.

Reply:
It would be valuable to make fusion mandatory at a later time once the API is more stable.  Due to the current implementation, things are easier to debug with fusion turned off.  This is because currently module fusion is implemented using the same tracing code as quantization.

Reply:
ok, makes sense